# HG changeset patch
# User Michael Tharp <mtharp@rpath.com>
# Date 1267054574 18000
# Node ID 60c10fe33f0c37eb0640e660b10b5df0574c8f6a
# Parent  e073911343d1ed3d631156d7e1cf23be3a32d148
Create auto databases as needed when doing PAUSE.

diff --git a/src/admin.c b/src/admin.c
--- a/src/admin.c
+++ b/src/admin.c
@@ -884,8 +884,14 @@
 		PgDatabase *db;
 		log_info("PAUSE '%s' command issued", arg);
 		db = find_database(arg);
-		if (db == NULL)
-			return admin_error(admin, "no such database: %s", arg);
+		if (db == NULL) {
+			db = register_auto_database(arg);
+			if (db == NULL) {
+				return admin_error(admin, "no such database: %s", arg);
+			} else {
+				slog_info(admin, "registered new auto-database for PAUSE: %s", arg);
+			}
+		}
 		if (db == admin->pool->db)
 			return admin_error(admin, "cannot pause admin db: %s", arg);
 		db->db_paused = 1;
